File: JAVA/test/DequeTest.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
import DataStructures.Exceptions.DataStructuresExceptions;
import DataStructures.Deques.Deque;
import DataStructures.Iter.Iterator;
/**
 *
 * @author argshub
 */
public class DequeTest {
    
    
    public static void main(String arg[]) {
        try {
            
            Deque<Integer> deque =  new Deque();
            Iterator iterator;
            deque.insertFront(100);
            deque.insertFront(200);
            deque.insertBack(300);
            deque.insertBack(400);
            
            
            iterator = deque.iterator();
            
            while(iterator.hasNext()) System.out.printf("%d\t", iterator.next());

            deque.removeFront();
            deque.removeBack();
            deque.removeFront();
            deque.removeBack();
            
        } catch(DataStructuresExceptions exceptions) {
            System.out.println(exceptions.getMessage());
        }
    }
}

File: CPP/DequeTest.cpp
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/* 
 * File:   DoublyLinkedListTest.cpp
 * Author: argshub
 *
 * Created on February 15, 2018, 7:00 PM
 */

#include <iostream>
#include "ds/deque.h"
#include "ds/exceptions.h"

using namespace std;

/*
 * 
 */
int main(int argc, char** argv) {
    try {
        
        Deque<int> deque;
        deque.insertFront(100);
        deque.insertFront(200);
        deque.insertBack(300);
        deque.insertBack(400);

        cout << deque.front() << "\n";
        cout << deque.back() << "\n";

        deque.removeFront();
        deque.removeBack();

        cout << deque.front() << "\n";
        cout << deque.back() << "\n";

        deque.removeFront();
        deque.removeBack();
        
    } catch(Exceptions exceptions) {
        cout << exceptions.get_message();
    }
    return 0;
}

